subsys/spm/Kconfig

```diff
@@ -48,6 +48,17 @@ menuconfig IS_SPM
 	select TRUSTED_EXECUTION_SECURE
 
 if IS_SPM
+
+config SPM_DEBUG
+	bool "SPM debug mode"
+	default y if NO_OPTIMIZATIONS
```

Comment:
Why should the SPM debugging be automatically selected when we build with NO_OPTIMIZATIONS, and/or DEBUG_OPTIMIZATIONS?

Reply:
@lemrey https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/964#issuecomment-509179787 wanted this as a default behavior.

It sort of makes sens if you think of it that you would default to debugging when doing a no-opt/debug-opt build. I don't know if there is a use case where you would do no-opt/debug-opt and don't want to debug. The only two I can think of is if you are doing a timing critical operation and don't want the compiler to change your code paths.  or security focused build. Is this correct? If so maybe it shouldn't be the default.